### PR TITLE
Members only (secret_society)

### PR DESCRIPTION
**Pull request by ajkacca457**
Secret society application contains the following features :
- Sign up new members
- New member can create secret posts
- No one can see the author's name and post description until they sign up or sign in.
- Hide post authors when not logged in
- Log in or Logout
- Members can only edit or delete their own posts.

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @ajkacca457 
### You did well in this project :clap: However, you have a few issues to fix:-
- [ ] It creates the application routes using `resources :posts, only: [:new, :create, :index]`
- [ ] Repeated components in the views can be put in the helpers ie posts/index.html.erb
### Make the changes and submit for another code review. Good luck.
- Happy coding!

**Comment by ajkacca457** [comment]
@rOluochKe Thank you very much for your review. It was very helpful in case of improving my project. Based on your review I have updated the routes, got read of some unnecessary view files. I have also created a separate partial for repeated code so that index view file is not overcrowded. I hope that my changes will meet the requirements.

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<p id="notice"class="text-center my-2 text-danger py-2 font-weight-bold"><%= notice %></p>
+
+<div class="container w-75 position-relative">
+
+<div class="titlecont bg-info d-flex px-2 py-2">
+<% if user_signed_in? %>
+<h3 class="text-white py-2">All the secrets of members</h3>
+<% else %>
+<h3 class="text-white py-2">Log In or Sign Up to see all the secrets in details</h3>
+<% end %>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb:10]: Kindly remove logic from views, implement them into view helpers

File: app/views/posts/_card.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<% @posts.each do |post| %>
+<div class="card text-left mt-2">
+  <div class="card-header bg-secondary text-white">
+  <% if user_signed_in? %>
+  <h5><%= post.user.name %></h5>
+  <% else %>
+  <h5>Annoynymous</h5>
+  <% end %>
```
Comment (maelfosso) [inline app/views/posts/_card.html.erb:8]: Kindly remove logic from views, implement them into view helpers

**Review by maelfosso — APPROVED** [review]
# Congratulations 👏🏾 

Thank you for implementing the suggested changes.
You can merge it now 👍🏾 

But before you do it, please, implement the changes from the comments left below.

Thanks and Good continuation 🤝 

Mael FOSSO

**Pull request merged**